returnn/datasets/cached.py

```diff
@@ -86,8 +86,12 @@ def init_seq_order(self, epoch=None, seq_list=None, seq_order=None):
     old_index_map = self._index_map[:]
     self._index_map = range(len(seq_index))  # sorted seq idx -> seq_index idx
 
-    if (isinstance(seq_index, numpy.ndarray) and numpy.array_equal(self._seq_index, seq_index)
-        or self._seq_index == seq_index) and self.start_cache_initialized:
+    if isinstance(seq_index, numpy.ndarray):
+        seq_index_unchanged = numpy.array_equal(self._seq_index, seq_index)  # equal operator would compare element-wise
```

Comment:
Indentation is wrong.

Reply:
Do you know how? I wonder that PyCharm does not warn about this. Or maybe it just needs to be enabled?

Reply:
I just checked why it does not do that, but the PyCharm inspector for `inconsistent indentation` only checks tabs vs spaces, but not the indentation level. If I move something by 4 I get no PyCharm warning.

Reply:
And the PEP8 inspector can only check for 4-space indents, we disabled this specifically to allow for a 2 space indentation.

Reply:
The indentation check is disabled here: https://github.com/rwth-i6/returnn/blob/master/tests/PyCharm.idea/inspectionProfiles/Project_Default.xml#L45

Reply:
Oops, fixed.
I also have checking for indentation disabled in my editor because of RETURNN. 😬

Reply:
@JackTemaki This is actually not used by our test case. It runs the PEP8 check independently. See here: https://github.com/rwth-i6/returnn/blob/0c6d863f4a4013acdd933ae21ae00cbbfd7309cd/tests/pycharm-inspect.py#L492

Reply:
Maybe when we add `--indent-size=2`, we don't need to exclude `E111,E114` and it works?

Reply:
I started draft PR #931 to test this.